### PR TITLE
Generic select

### PR DESCRIPTION
**Pull request by nicopaul**
This change is [Reviewable](https://reviewable.io/reviews/testit-webtester/webtester-core/68)

### CODE REVIEW

File: webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    protected String getHTMLFilePath() {
+        return "html/pageobjects/genericSelect.html";
+    }
+
+    /* get all option texts */
+
+    @Test
+    public final void testThatListWithOptionTextsIsReturnedCorrectly_singleSelect() {
+        List<String> texts = page.singleSelect.getAllOptionTexts();
+        assertThat(texts.get(0).equals("one"), is(true));
```
Comment (slu-it) [inline webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java]: `assertThat(texts.get(0).equals("one"), is(true));` >> `assertThat(texts.get(0), is("one"));`
(for all tests in this class)

File: webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        List<String> texts = page.multiSelect.getAllOptionTexts();
+        assertThat(texts.get(0).equals("one"), is(true));
+        assertThat(texts.get(1).equals("two"), is(true));
+        assertThat(texts.get(2).equals("three"), is(true));
+    }
+
+    @Test
+    public final void testThatListWithOptionTextsIsReturnedCorrectly_emptySelect() {
+        List<String> optionTexts = page.emptySelect.getAllOptionTexts();
+        assertThat(optionTexts.isEmpty(), is(true));
```
Comment (slu-it) [inline webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java]: `assertThat(optionTexts.isEmpty(), is(true));` >> `assertThat(optionTexts, is(empty()));`

File: webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        assertThat(values.get(0).equals("1"), is(true));
+        assertThat(values.get(1).equals("2"), is(true));
+        assertThat(values.get(2).equals("3"), is(true));
+    }
+
+
+    @Test
+    public final void testThatListWithOptionValuesIsReturnedCorrectly_emptySelect() {
+        List<String> optionValues = page.emptySelect.getAllOptionValues();
+        assertThat(optionValues.isEmpty(), is(true));
```
Comment (slu-it) [inline webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java]: see above comment

File: webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java (hunk 2 above)
Comment (nicopaul) [inline webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java]: Which import do i have to use? 
java.util? But isn't Optional.empty()  java8?

Comment (slu-it) [inline webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java]: `org.hamcrest.Matchers.empty`

All Hamcrest matchers have a factory method within the `Matchers` class.

Comment (nicopaul) [inline webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java]: oh just overlooked it
thanks

**Pull request merged**